app/contracts/user_preferences/params_contract.rb

```diff
@@ -28,16 +28,20 @@
 
 module UserPreferences
   class ParamsContract < ::ParamsContract
+    include RequiresEnterpriseGuard
+    self.enterprise_action = :date_alerts
+    self.enterprise_condition = ->(*) do
+      (NotificationSetting.date_alert_settings & notifications.flat_map(&:keys)).any?
```

Comment:
Can you extract it in a method so that it explains the intent? something like `date_alert_notifications_provided?`